src/index/hgraph_index.cpp

```diff
@@ -95,6 +95,11 @@ tl::expected<std::vector<int64_t>, Error>
 HGraphIndex::add(const DatasetPtr& data) {
     std::vector<int64_t> failed_ids;
     try {
+        auto base_dim = data->GetDim();
+        CHECK_ARGUMENT(base_dim == dim_,
+                       fmt::format("base.dim({}) must be equal to index.dim({})", base_dim, dim_));
+        CHECK_ARGUMENT(data->GetFloat32Vectors() != nullptr, "base.float_vector is nullptr");
```

Comment:
data.float_vector is nullptr

Reply:
actually base means "data of the database"